pages/itinerary.js

```diff
@@ -4,8 +4,34 @@ import Link from "next/link";
 import { uid } from "uid";
 import { format } from "date-fns";
 import Carousel from "@/components/Carousel";
+import { useState, useEffect } from "react";
 
 export default function Itinerary({ trips, sights, setSights }) {
+  const [sortedSights, setSortedSights] = useState([]);
+
+  useEffect(() => {
+    const updatedSortedSights = sights.slice().sort((a, b) => {
+      if (a.plannedTime === "") {
+        if (b.plannedTime === "") {
+          return 0;
+        } else {
+          return 1;
+        }
+      } else {
+        if (b.plannedTime === "") {
+          return -1;
+        }
+        if (a.plannedTime < b.plannedTime) {
+          return -1;
+        }
+        if (a.plannedTime > b.plannedTime) {
+          return 1;
+        }
+      }
+    });
+    setSortedSights(updatedSortedSights);
+  }, [sights]);
```

Comment:
First of all, great effort on the sorting. I think there is room to improve it slightly, but that is not want i want to focus on here. 

Instead i think we can remove the `useState` and the `useEffect` all together. The component will re-render every time the props change, so we can create a variable directly in the component based on the props. i.e.

```suggestion
const sortedSights = sights.slice().sort((a, b) => {
  // Sorting logic here
});

```